experimental/vjn/yarpl/main/main.cc

```diff
@@ -0,0 +1,80 @@
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+#include <reactivestreams/subscriber.h>
+#include <yarpl/flowable.h>
+#include <yarpl/scheduler.h>
+
+namespace {
+
+using reactivestreams::Subscriber;
+
+auto make_subscriber(int state) {
+  return Subscriber<int>::create(
+      [state](int i) mutable {
+        std::cout << "  state: " << ++state << ", i = " << i << std::endl;
+      },
+      [](const std::exception_ptr eptr) {
+        try {
```

Comment:
to avoid expensive try ... catch .. rethrow .. observe type, folly provides folly::exception_wrapper which does the same thing, but much cheaper.